chainer/training/extensions/progress_bar.py

```diff
@@ -44,10 +43,10 @@ def __call__(self, trainer):
         # initialize some attributes at the first call
         if training_length is None:
             t = trainer.stop_trigger
-            if not isinstance(t, triggers.IntervalTrigger):
+            if not hasattr(t, '_get_training_length'):
```

Comment:
This method is used as an interface of trigger, so I think it's better not to prefix it by `_`. How about using `get_training_length`?

Reply:
I also feel it's more natural to let duck-typing check the code here. I mean just removing this if-then-raise block. What do you think? It will show the trigger type and the cause of the error (`'FooTrigger' object has no attribute 'get_training_length'`), which are more informative for users.

Reply:
> This method is used as an interface of trigger
I understand! I remove the prefix.

Reply:
> I also feel it's more natural to let duck-typing check the code here.

Thanks for the comment! I agree with your suggestion.
It would be better to show informations